omod/src/main/web/dashboardwidgets/latestobsforconceptlist/latestobsforconceptlist.controller.js

```diff
@@ -15,39 +14,35 @@ export default class LatestObsForConceptListController {
         this.maxAgeInDays = this.widgetsCommons.maxAgeToDays(this.config.maxAge);
 
         // Remove whitespaces
-        this.config.concepts = this.config.concepts.replace(/\s/g,'');
-
-        let concepts = this.config.concepts.split(",");
-        for (let i = 0; i < concepts.length; i++) {
-            let concept = concepts[i];
-            // Fetch last obs for concept
-            this.openmrsRest.list('obs', {
-                patient: this.config.patientUuid,
-                v: 'full',
-                concept: concept
-            }).then((resp) => {
-                // Don't add more items if max concept count is reached
-                if (this.obs.length < this.maxConceptCount && resp.results.length > 0) {
-                    let obs = resp.results[0];
-                    // Don't add obs older than maxAge
-                    if (angular.isUndefined(this.maxAgeInDays) || this.widgetsCommons.dateToDaysAgo(obs.obsDatetime) <= this.maxAgeInDays) {
-                        // Add last obs for concept to list
-
-                        if (['8d4a505e-c2cc-11de-8d13-0010c6dffd0f',
-                                '8d4a591e-c2cc-11de-8d13-0010c6dffd0f',
-                                '8d4a5af4-c2cc-11de-8d13-0010c6dffd0f'].indexOf(obs.concept.datatype.uuid) > -1) {
-                            //If value is date, time or datetime
-                            var date = this.$filter('date')(new Date(obs.value), this.config.dateFormat);
-                            obs.value = date;
-                        } else if (angular.isDefined(obs.value.display)) {
-                            //If value is a concept
-                            obs.value = obs.value.display;
-                        }
-
-                        this.obs.push(obs);
+        let concept_list = this.config.concepts.replace(/\s/g,'');
+
+        // Fetch last obs for the list of concepts
+        this.openmrsRest.list('latestobs', {
+            patient: this.config.patientUuid,
+            v: 'full',
+            concept: concept_list
+        }).then((resp) => {
+            // Process the results from the list of concepts as not all of them may have data
+            for (let i = 0; i < resp.results.length; i++) {
```

Comment:
Did you intentionally ignore the this.maxConceptCount check?

Reply:
@dkayiwa That feature was not in the specification https://wiki.openmrs.org/display/docs/Patient+Summary+Widget+Documentation#PatientSummaryWidgetDocumentation-conceptlistLatestObsForConceptList

Reply:
Does that mean the original code was some how buggy?

Reply:
@dkayiwa It was never documented in the spec that was being used to develop the widget

Reply:
Wouldn't removing it have effects on those who were depending on it?
Can't we improve the performance without altering existing behaviour?

Reply:
Let me add back the check, but I am wondering how they are using it when its not documented

Reply:
Am not sure. I do not even know if any one is using it. Am just cautious of introducing unintended effects in a ticket which is meant to simply improve performance.

Reply:
@dkayiwa  I think we should remove the check as its undocumented behavior, which is also not configurable as part of the widget app. Why should the widget limit the maxiumum number of concepts with no way to change it

Reply:
Am not against removing it. But am very much against doing so using a ticket whose only purpose is improving performance. This should not have any side effects other than that.

Reply:
NB: This seems similar to https://issues.openmrs.org/browse/RA-1531, though that's about the ObsAcrossEncounters widget.

Maybe the right thing to do here is file a new ticket and attach it to this PR as "also fixes"?

Reply:
Having one commit to address two tickets is not a good practice.

Reply:
It's not uncommon for big overhauls, refactors, etc. to fix a handful of bugs along the way, just by replacing old code with a totally new and simpler approach. This kind of thing happens all the time, and is part of the point of a refactor.

You're right that it's a bit messy in terms of process. But I think it's worth it not to be intentionally re-implementing bugs.

Reply:
Even reviewing code that does more than what a ticket was meant for, becomes confusing to the reviewer. It has always been our practice that if there is a bug, one creates a new ticket of type bug, and then raises a pull request for it. The ticket includes a reasonable amount of documentation about the bug and why it should be fixed this way. Looking at this commit yourself, after a while, you can take some time before remembering how all these changes are related to the the performance improvement.

Reply:
I have updated the PR to include the feature limiting the maximum number of concepts on the widget to 10